### PR TITLE
🌱 Bump k8s to v1.31.0 and drop caBundle from CRDs to support Kubernetes 1.31

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
**What this PR does / why we need it**:
Bump k8s to v1.31.0 and Drop caBundle from CRDs to support Kubernetes 1.31. 
ref: https://github.com/kubernetes-sigs/cluster-api/pull/10972

In addition this PR is also the fix in `ip_reuse` test, in the test when doing `k8s upgrade` from `1.30.0 to 1.31.0` we were only changing k8s version in `MachineDeployment`, which is not enough we have to update `Metal3MachineTemplate` too with the new image, and sha. It is fixed now in this PR.

### CODE REVIEW

File: scripts/environment.sh [hunk 1]
```diff
@@ -49,7 +49,7 @@ else
 fi
 
 export FROM_K8S_VERSION=${FROM_K8S_VERSION:-"v1.29.0"}
```
Comment (smoshiur1237) [inline scripts/environment.sh:51]: should we change it to v1.30.0?

**Comment by adilGhaffarDev** [comment]
/hold
wait until node images are available in artifactory

**Comment by adilGhaffarDev** [comment]
tests passed after the fix in `ip_reuse` test, in the test when doing `k8s upgrade` from `1.30 to 1.31-rc.0` we were only changing k8s version in `MachineDeployment`, which is not enough we have to update `Metal3MachineTemplate` too with the new image, and sha. It is fixed now. ref to passed test: https://jenkins.nordix.org/view/Metal3/job/metal3-ubuntu-e2e-feature-test-main-features/8/consoleFull

Now running remaining tests.

/test metal3-ubuntu-e2e-feature-test-main-pivoting
/test metal3-ubuntu-e2e-feature-test-main-remediation
/test metal3-centos-e2e-feature-test-main-features

these should be enough to say that `k8s 1.31-rc.0` uplift is working. I will run `e2e-integeration` tests once these are passed.

**Comment by Sunnatillo** [comment]
/retitle 🌱 Bump k8s to v1.31.0 and drop caBundle from CRDs to support Kubernetes 1.31

**Comment by adilGhaffarDev** [comment]
seems like a flake, bmhs were provisioned evetually
/test metal3-centos-e2e-feature-test-main-features

**Comment by adilGhaffarDev** [comment]
tests passed, @Sunnatillo @kashifest @smoshiur1237 please check.

**Comment by kashifest** [comment]
> tests passed, @Sunnatillo @kashifest @smoshiur1237 please check.

what was the issue

**Review by smoshiur1237 — COMMENTED** [review]
/approve

**Comment by adilGhaffarDev** [comment]
> what was the issue

it was in ip_reuse test, it is explained here: https://github.com/metal3-io/cluster-api-provider-metal3/pull/1866#issuecomment-2284011074

**Pull request merged**